examples/material_gallery/test/smoke_test.dart

```diff
@@ -8,55 +8,122 @@ import 'package:test/test.dart';
       tester.pump(); // triggers a frame
 
-      // Try loading Weather demo
-      tester.tap(find.text('Demos'));
-      tester.pump();
-      tester.pump(const Duration(seconds: 1)); // wait til it's really opened
+      // Expand the demo category submenus.
+      for(String category in demoCategories.reversed) {
+        tester.tap(find.text(category));
+        tester.pump();
+        tester.pump(const Duration(seconds: 1)); // Wait until the menu has expanded.
+      }
 
-      tester.tap(find.text('Weather'));
-      tester.pump();
-      tester.pump(const Duration(seconds: 1)); // wait til it's really opened
-
-      // Go back
-      Finder backButton = find.byElement((Element element) {
-        Widget widget = element.widget;
-        if (widget is Tooltip)
-          return widget.message == 'Back';
-        return false;
-      });
-      expect(tester, hasWidget(backButton));
-      tester.tap(backButton);
-      tester.pump(); // start going back
-      tester.pump(const Duration(seconds: 1)); // wait til it's finished
-
-      // Open menu
-      Finder navigationMenu = find.byElement((Element element) {
-        Widget widget = element.widget;
-        if (widget is Tooltip)
-          return widget.message == 'Open navigation menu';
-        return false;
-      });
-      expect(tester, hasWidget(navigationMenu));
-      tester.tap(navigationMenu);
-      tester.pump(); // start opening menu
-      tester.pump(const Duration(seconds: 1)); // wait til it's really opened
+      final List<double> scrollDeltas = new List<double>();
+      double previousY = tester.getTopRight(find.text(demoCategories[0])).y;
+      for(String name in demoNames) {
+        final double y = tester.getTopRight(find.text(name)).y;
+        scrollDeltas.add(previousY - y);
+        previousY = y;
+      }
+
+      // Launch each demo and then scroll that item out of the way.
+      for(int i = 0; i < demoNames.length; i++) {
+        final String name = demoNames[i];
+        print("$name");
+        smokeDemo(tester, name);
+        tester.scroll(find.text(name), new Offset(0.0, scrollDeltas[i]));
+        tester.pump();
+      }
+
+      Finder navigationMenuButton = findNavigationMenuButton(tester);
+      expect(tester, hasWidget(navigationMenuButton));
+      tester.tap(navigationMenuButton);
+      tester.pump(); // Start opening drawer.
+      tester.pump(const Duration(seconds: 1)); // Wait until it's really opened.
 
       // switch theme
       tester.tap(find.text('Dark'));
       tester.pump();
-      tester.pump(const Duration(seconds: 1)); // wait til it's changed
+      tester.pump(const Duration(seconds: 1)); // Wait until it's changed.
 
       // switch theme
       tester.tap(find.text('Light'));
       tester.pump();
-      tester.pump(const Duration(seconds: 1)); // wait til it's changed
+      tester.pump(const Duration(seconds: 1)); // Wait until it's changed.
```

Comment:
We could "pre-smoke" the app by verifying that all of the expected GalleryHome menu items can be found. I'll make that update as part of the PR noted here https://github.com/flutter/flutter/pull/3446#discussion_r60595139